sympy/combinatorics/fp_groups.py

```diff
@@ -514,6 +514,138 @@ def elements(self):
         P, T = self._to_perm_group()
         return T.invert(P._elements)
 
+def subgroup_quotient(G, H, parent_group=None, homomorphism=False):
+    '''
+    Compute the quotient group G/H.
+    The quotient group is computed on a new FreeGroup when
+    `G` list of the elements of parent_group.
+
+    Arguments
+    =========
+    G -- A list of `FreeGroupElement`s or an `FpGroup`.
+    H -- A list of `FreeGroupElement`s or an `FpGroup`.
+    parent_group -- A group specified when `G` and `H` are given by a list of generators.
+    homomorphism -- Return a homomorphism whenever `homomorphism` = True
+
+    Returns
+    =======
+    * When `homomorphism` = True, quotient group along with the homomorphism
+    from the quotient to the parent_group whose image is isomorphic to G
+    when G is generated any set of elements of the parent_group
+    * Only the quotient group in all other cases.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.fp_groups import FpGroup
+    >>> from sympy.combinatorics.free_groups import free_group
+    >>> from sympy.combinatorics.fp_groups import subgroup_quotient, maximal_abelian_quotient
+
+    >>> F, x, y = free_group("x, y")
+    >>> f = FpGroup(F, [x**2, y**3, (x*y)**4])
+    >>> T = subgroup_quotient(f, [x, y])
+    >>> H = f.subgroup([x, y])
+    >>> T.order() == f.order()/H.order()
+    True
+
+    >>> f = FpGroup(F, [x**2, y**3, (x*y)**4])
+    >>> T = subgroup_quotient(f, [x*y**2*x*y, y**2*x*y*x, y**-1])
+    >>> H = f.subgroup([x*y**2*x*y, y**2*x*y*x, y**-1])
+    >>> T.order() == f.order()/H.order()
+    True
+
+    >>> G = [x, y]
+    >>> H = [x*y**2*x*y, y**2*x*y*x, x*y*x]
+    >>> T = subgroup_quotient(G, H, parent_group=f)
+    >>> G = f.subgroup(G)
+    >>> H = f.subgroup(H)
+    >>> T.order() == G.order()/H.order()
+    True
+
+    '''
+    def _get_pres(parent_group, F):
+        from sympy.combinatorics.homomorphisms import homomorphism
+        if isinstance(F, list):
+            K, T = parent_group.subgroup(F, homomorphism=True)
+            f_gens = T(K.generators)
+            f_rels = T(K.relators)
+        else:
+            f_gens = F.generators
+            f_rels = F.relators
+        return f_gens, f_rels
+
+    def _check(F):
+        if ((isinstance(F, list) and not all(elem in free_group for elem in F))
+        or (isinstance(F, FpGroup) and not (F.free_group == free_group))):
+            raise ValueError("The group elements must belong to the parent group")
+
+    # If no parent group is specified,
+    # G is set to the parent `FpGroup`
+    if not parent_group:
+        if isinstance(G, list):
+            raise ValueError("The parent_group must be"
+                            "defined when the group is a list")
+        parent_group = G
+
+    free_group = parent_group.free_group
```

Comment:
We don't need this statement - we always use the free group of `G`

Reply:
Still the case

Reply:
But, it has to be defined before the `_check` function is called.